app/src/main/java/org/mozilla/fenix/settings/biometric/BiometricPromptPreferenceFragment.kt

```diff
@@ -0,0 +1,123 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.biometric
+
+import android.app.Activity.RESULT_OK
+import android.app.KeyguardManager
+import android.content.Context
+import android.content.Intent
+import android.view.View
+import androidx.core.content.getSystemService
+import androidx.preference.Preference
+import androidx.preference.PreferenceFragmentCompat
+import mozilla.components.support.base.feature.ViewBoundFeatureWrapper
+import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.settings.requirePreference
+
+/**
+ * Helper for creating and implementing the [BiometricPromptFeature]. Currently used
+ * for logins and credit cards.
+ */
+abstract class BiometricPromptPreferenceFragment : PreferenceFragmentCompat() {
+
+    internal val biometricPromptFeature = ViewBoundFeatureWrapper<BiometricPromptFeature>()
+
+    /**
+     * Gets the string to be used for [BiometricPromptFeature.requestAuthentication] prompting to
+     * unlock the device.
+     */
+    abstract fun unlockMessage(): String
+
+    /**
+     * Navigate when authentication is successful.
+     */
+    abstract fun navigateOnSuccess()
+
+    /**
+     * Show a warning to set up a pin/password when the device is not secured. This is only used
+     * when BiometricPrompt is unavailable on the device.
+     */
+    abstract fun showPinDialogWarning(context: Context)
+
+    /**
+     * There is a bug where while the biometric prompt is showing, you were able to quickly navigate
+     * so we are disabling the settings that navigate while authenticating.
+     * https://github.com/mozilla-mobile/fenix/issues/12312
+     *
+     * @param prefList a list of [Preference]s to toggle.
+     * @param enabled whether or not the preferences should be enabled.
+     */
+    fun togglePrefsEnabledWhileAuthenticating(prefList: List<Int>, enabled: Boolean) {
+        for (preference in prefList) {
+            requirePreference<Preference>(preference).isEnabled = enabled
+        }
+    }
+
+    /**
+     * Create a prompt to confirm the device's pin/password and start activity based on the result.
+     * This is only used when BiometricPrompt is unavailable on the device.
+     */
+    @Suppress("Deprecation")
+    abstract fun showPinVerification(manager: KeyguardManager)
+
+    /**
+     * Sets the biometric prompt feature.
+     *
+     * @param view the view that the prompt will be associate with.
+     * @param prefList a list of [Preference]s to toggle.
+     */
+    fun setBiometricPrompt(view: View, prefList: List<Int>) {
```

Comment:
Did we consider overriding `onViewCreated` to avoid having inherited fragment call this manually?

Reply:
Please provide a response before resolving a conversation if no action was taken. I wouldn't be leaving a comment if there was no merit to it. Best case scenario would be to think about it and decide if changed were warranted or provide some thoughtful feedback to comment. If it's decided we wanted to do this as a followup, leaving the comment unresolve and linking to it back in the followup ticket would also been a better option.